### PR TITLE
WT-11459 Disable background compaction server for inmem and readonly databases

### PR DESCRIPTION
**Pull request by etienneptl**

### CODE REVIEW

**Review by kommiharibabu — APPROVED** [review]
LGTM

File: test/suite/test_compact09.py [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+# MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.
+# IN NO EVENT SHALL THE AUTHORS BE LIABLE FOR ANY CLAIM, DAMAGES OR
+# OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE,
+# ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR
+# OTHER DEALINGS IN THE SOFTWARE.
+
+import wiredtiger, wttest
+
+# test_compact09.py
+# Test compaction for in-memory and readonly databases.
```
Comment (mm-ng) [inline test/suite/test_compact09.py:32]: Maybe this comment could be changed to make it clearer that the compact API is not compatible with in-memory and readonly databases?

Comment (etienneptl) [reply test/suite/test_compact09.py]: Updated in b6e861c35de528d2d988411d354a54ffe3d5adcf

**Review by Sean04 — APPROVED** [review]
LGTM

**Comment by etienneptl** [comment]
evergreen merge

**Pull request merged**